pkg/juicefs/juicefs.go

```diff
@@ -806,13 +806,33 @@ func (j *juicefs) SetQuota(ctx context.Context, secrets map[string]string, jfsSe
 		cmdArgs = []string{config.CliPath, "quota", "set", secrets["name"], "--path", quotaPath, "--capacity", strconv.FormatInt(cap, 10)}
 	}
 	klog.Infof("SetQuota cmd: %s", strings.Join(cmdArgs, " "))
-	cmdCtx, cmdCancel := context.WithTimeout(ctx, 10*defaultCheckTimeout)
+	cmdCtx, cmdCancel := context.WithTimeout(ctx, 2*defaultCheckTimeout)
 	defer cmdCancel()
 
 	var res []byte
 	var err error
 	if jfsSetting.IsCe {
-		res, err = j.Exec.CommandContext(cmdCtx, config.CeCliPath, args...).CombinedOutput()
+		done := make(chan struct{})
+		go func() {
+			// ce cli will block until quota is set
+			res, err = j.Exec.CommandContext(context.Background(), config.CeCliPath, args...).CombinedOutput()
+			close(done)
+		}()
+		select {
+		case <-cmdCtx.Done():
+			go func() {
+				klog.Warningf("quota set timeout, runs in background")
+				<-done
+				if err := wrapSetQuotaErr(string(res), err); err != nil {
```

Comment:
There is no need to start a new goroutine to collect logs. You can print logs directly in the `set qupta` goroutine.

Reply:
code is polished